envoy/types/virtualhost.go

```diff
@@ -0,0 +1,43 @@
+package types
+
+import (
+	"fmt"
+
+	route "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
+)
+
+type VirtualHost struct {
+	*route.VirtualHost
+}
+
+func NewVirtualHost(name string) *VirtualHost {
+	return &VirtualHost{
+		&route.VirtualHost{
+			Name:    name,
+			Domains: []string{},
+		},
+	}
+}
+
+func (v *VirtualHost) AddDomain(domain string) {
+	v.Domains = append(v.Domains, domain)
+}
+
+func (v *VirtualHost) AddRoute(r *route.Route) error {
+	routeExists := func(name string, routes []*route.Route) bool {
```

Comment:
This function routeExists var will be allocated each time we enter AddRoute, and it doesn't require the access to the inner AddRoute variables - i.e. is not justified for the closure definition here, better to move it outside as it was before.

Reply:
No it's not. There's actually no such thing as allocating a function in Go - it's just a syntactic sugar. Compiler will either inline this function or make a package-level one with surrounding function's name as a prefix.